addons/microsoft_outlook/models/microsoft_outlook_mixin.py

```diff
@@ -78,8 +78,12 @@ def open_microsoft_outlook_uri(self):
         if not self.env.user.has_group('base.group_system'):
             raise AccessError(_('Only the administrator can link an Outlook mail server.'))
 
-        if not self.use_microsoft_outlook_service or not self.is_microsoft_outlook_configured:
-            raise UserError(_('Please configure your Outlook credentials.'))
+        if not self.use_microsoft_outlook_service or not self.is_microsoft_outlook_configured or not self.smtp_user:
```

Comment:
fyi, I meant more like a `@constrains` in the settings but this should be more than enough actually.

Reply:
Tried adding something `@contrains` it but couldn't make it work with the Connect Outlook Account action. I.e. couldn't find the right logical conditions to prompt the user to fill a username (behaved strangely ecayse `open_microsoft_outlook_uri` gets called as an action which triggers the current (new) record form to be saved and fail the OAuth). But this might be mostly me not going the right way about it.

Simply adding the extra step before the action is called seemed the cleanest and easiest, as it doesn't commit the (new) record before a `smtp_user` is filled.

Additionally I just tested the workflow with the gmail OAuth, and the issue is the same: if no username is defined, the token string doesn't get generated and you won't be able to send emails. It seems that the gmail mixin uses the same logic, so I guess I can adapt the same error message to the gmail module. Will try locally if installing both modules with the proposed changes creates any unforeseen effects.

Reply:
Added the user error in both modules (`google_gmail` and `microsoft_outlook`). Functionally I tested it locally, and if `smtp_user` is empty, the user gets the error before the OAuth login page is loaded.